server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -103,21 +106,25 @@ public RemoteFsTranslog(
         ThreadPool threadPool,
         BooleanSupplier startedPrimarySupplier,
         RemoteTranslogTransferTracker remoteTranslogTransferTracker,
-        RemoteStoreSettings remoteStoreSettings
+        RemoteStoreSettings remoteStoreSettings,
+        Supplier<Version> minNodeVersionSupplier
     ) throws IOException {
         super(config, translogUUID, deletionPolicy, globalCheckpointSupplier, primaryTermSupplier, persistedSequenceNumberConsumer);
         logger = Loggers.getLogger(getClass(), shardId);
         this.startedPrimarySupplier = startedPrimarySupplier;
         this.remoteTranslogTransferTracker = remoteTranslogTransferTracker;
-        fileTransferTracker = new FileTransferTracker(shardId, remoteTranslogTransferTracker);
+        boolean ckpAsTranslogMetadata = isCkpAsTranslogMetadata(minNodeVersionSupplier, blobStoreRepository);
```

Comment:
The minNode version check is being used only at the time of RemoteFsTranslog initialisation. I see an improvement where we can use the min version check until the time of upload.

Reply:
This will help to move the existing indexes to use the new approach. But fine if you can create an issue and get this done in a follow up.